Sources/SourceKitLSP/Swift/GeneratedInterfaceDocumentURLData.swift

```diff
@@ -67,7 +67,14 @@ package struct GeneratedInterfaceDocumentURLData: Hashable, ReferenceURLData {
     self.moduleName = moduleName
     self.groupName = groupName
     self.sourcekitdDocumentName = sourcekitdDocumentName
-    self.buildSettingsFrom = primaryFile
+    if let referenceDocumentURL = try? ReferenceDocumentURL(from: primaryFile),
+       let fileURL = referenceDocumentURL.buildSettingsFile.fileURL {
+      self.buildSettingsFrom = DocumentURI(fileURL)
+    } else if let fileURL = primaryFile.fileURL {
+      self.buildSettingsFrom = DocumentURI(fileURL)
+    } else {
+      self.buildSettingsFrom = primaryFile
+    }
```

Comment:
@ahoppen 

make sure buildSettingsFrom can be only an actual file otherwise

it could be 
sourcekit-lsp://xxx?buildSettingsFrom=A
A=sourcekit-lsp://xxx?buildSettingsFrom=B

NOTE: this happens if we enable navigation inside swiftinterface in this PR